crates/re_viewer_context/src/viewer_context.rs

```diff
@@ -82,21 +90,22 @@ impl<'a> ViewerContext<'a> {
         self.rec_cfg.time_ctrl.read().current_query()
     }
 
-    /// Returns whether the given tree has any data logged in the current timeline,
-    /// or has any timeless messages.
-    pub fn tree_has_data_in_current_timeline(&self, tree: &EntityTree) -> bool {
-        let top_time_histogram = &tree.subtree.time_histogram;
-        top_time_histogram.has_timeline(self.rec_cfg.time_ctrl.read().timeline())
-            || top_time_histogram.num_timeless_messages() > 0
-    }
-
-    /// Returns whether the given component has any data logged in the current timeline.
-    pub fn component_has_data_in_current_timeline(
+    /// The current time query, based on the current time control and an `entity_path`
+    ///
+    /// If the blueprint timeline is visible, and the `entity_path` is on the blueprint
+    /// timeline, then the blueprint timeline is used. Otherwise, return the regular
+    /// timeline.
+    pub fn current_query_for_entity_path(
```

Comment:
This feels quite inelegant.

I guess the alternative would be for the current `Item` selection to know what store it is on, i.e. that we don't select only an EntityPath`, but an `StoreId + EntityPath` pair. I guess that's more work though, but did you consider it?

Reply:
Yeah, the need for this bit made me sad.  I considered 2 options and both were further reaching than I wanted to tackle here:
 1. As you suggest: adding it to the Selection context and then plumbing the information down into every data_ui element.
 2. Introduce a GloballyQualifiedEntityPath (or equivalent name) which tracks StoreId + Path, and migrating many of our usages of EntityPath (specifically Selections) to use GQEP instead.
 
(2) is actually be preference at the moment because GQEP's with blueprint store-ids would also be helpful with the implementation of generalized hierarchical component-overrides that combine both log-centric (e.g. log-recursive) and blueprint-sourced overrides.